### PR TITLE
Feature/je 116 expenses

### PR DESCRIPTION
**Pull request by tuj**
* Adds expenses to invoice entries.
* Refactored invoiceEntry filter and tables to be component based.
* Added expense entity.
* Set isBilled for Worklog and Expense entities when the invoice is recorded.

### CODE REVIEW

File: bundles/Billing/Resources/assets/components/ExpenseSelect.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            let startFilterTimestamp = filterValues.startDateFilter.getTime();
+
+            if (startFilterTimestamp > worklogUpdatedTimestamp) {
+                return false;
+            }
+        }
+
+        if (filterValues.endDateFilter !== null && filterValues.endDateFilter !== '') {
+            let endDate = filterValues.endDateFilter;
+            endDate.setHours(23, 59, 59);
```
Comment (rimi-itk) [inline bundles/Billing/Resources/assets/components/ExpenseSelect.js]: This looks fishy. What about the second between 23:59:59 and 00:00:00? `less than` (but not equal to) in the following comparison should not require this hack.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
A small timing issue must be addressed.

File: bundles/Billing/Resources/assets/components/ExpenseSelect.js (hunk 1 above)
Comment (tuj) [reply bundles/Billing/Resources/assets/components/ExpenseSelect.js]: Point taken :)

**Review by rimi-itk — APPROVED** [review]
LGTM.

**Pull request merged**